tests/structured_conf/test_structured_config.py

```diff
@@ -864,6 +864,85 @@ def test_create_untyped_list(self, class_type: str) -> None:
         assert cfg.list == [1, 2]
         assert cfg.opt_list is None
 
+    class TestInstantiateStructuredConfigs:
+        @pytest.fixture
+        def module(self, class_type: str) -> Any:
+            module: Any = import_module(class_type)
+            return module
+
+        def round_trip_to_container(self, input_data: Any) -> Any:
+            serialized = OmegaConf.create(input_data)
+            round_tripped = OmegaConf.to_container(
+                serialized, instantiate_structured_configs=True
+            )
+            return round_tripped
+
+        def test_basic(self, module: Any) -> None:
+            user = self.round_trip_to_container(module.User())
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name is MISSING
+            assert user.age is MISSING
+
+            user = self.round_trip_to_container(module.User("Bond", 7))
+            assert isinstance(user, module.User)
+            assert type(user) is module.User
+            assert user.name == "Bond"
+            assert user.age == 7
+
+        def test_nested(self, module: Any) -> None:
+            data = self.round_trip_to_container({1: module.User()})
```

Comment:
I know you like using integers as dict key but it's hardly the most common pattern.
test with a string as the dict key.
```suggestion
            data = self.round_trip_to_container({"user": module.User()})
```

Reply:
This test is testing an object nested in a dict.
Add a test for an object nested in an object.